pkg/rpc/client.go

```diff
@@ -31,6 +31,9 @@ type ETHClient interface {
 	// between contract internal errors and the local chain being out of sync.
 	CodeAt(ctx context.Context, contract common.Address, blockNumber *big.Int) ([]byte, error)
 
+	// TransactionReceipt implements the geth backend DeployBackend interface.
+	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
```

Comment:
This is added so that the eth client monitors the completion of a transaction. The transaction gas used value is then saved in the db.